### ISSUE TITLE
Import faliure, AttributeError: 'NoneType' object has no attribute 'outputs'

### ISSUE THREAD

**Issue by thelabcat** [NoneGiven/MphRead#129]
I attempted to import UNIT2_RM8, "Docking Bay". The mesh imported when I ran the script, but was left with only vertex colors in use, and Python raised this error:

```
Traceback (most recent call last):
  File "/import_UNIT2_RM8.py", line 1101, in <module>
  File "/import_UNIT2_RM8.py", line 20, in import_dae
  File "/home/wilbur/Bin/MphRead-0.26.0.0-linux/modules/mph_common.py", line 160, in set_texture_alpha
    material.link_nodes(
  File "/home/wilbur/Bin/MphRead-0.26.0.0-linux/modules/mph_common.py", line 84, in material_link_nodes
    source.outputs[output],
    ^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute 'outputs'
```

**Comment by thelabcat** [comment]
The `material_link_nodes()` function is being passed an argument named "self". Side note, this argument should have a more recognizable name: This function isn't a method of whatever "self" is. Anyways, "self" has a method `get_node()`, which for some reason is returning NoneType in this case, and `material_link_nodes()` has no catch for that. When I put in a catch, script no longer raises an exception, but the output of the script unfortunately remains the same.

**Comment by NoneGiven** [comment]
What version of Blender are you using?

**Comment by thelabcat** [comment]
Latest stable Snap at the time.

**Comment by NoneGiven** [comment]
Tested with Blender 4.1.1. This isn't a problem with the script, and probably not a problem with MphRead. It appears to be that your Collada import can't find the texture images. The import then creates the materials without a "Base Image" node, causing the script to fail when it tries to update the shading nodes.

The folder with the `.dae` file should have a subfolder named `default` containing 21 images for UNIT2_RM8.

<img width="983" alt="image" src="https://github.com/NoneGiven/MphRead/assets/2163967/91ca7872-9797-49a4-98b9-cf6aecce05ee">

Confirm this is the case before importing. If it is, then I'll need some help investigating further because I can't reproduce the issue with the export done properly.

**Comment by thelabcat** [comment]
Oh, not all of them, but I see what you're getting at.
```
<image id="lambert41" name="lambert41">
	<init_from>default\10-10.png</init_from>
</image>
```
I'll correct the paths, and see if it works.

Tip: To get which direction the slash is supposed to go in python, use the builtin os library. The slash is stored as `os.sep` there.

**Comment by thelabcat** [comment]
Confirmed, that fixed the issue. So, this is another cross-platform concern.

By the way, it isn't needed now, but here's the catcher I made for the error:
```
def material_link_nodes(self, source, output, target, input):
    if (isinstance(source, str)):
        print("Selecting node", source, "by name")
        source = self.get_node(source)
    if (isinstance(target, str)):
        print("Selecting node", target, "by name")
        target = self.get_node(target)

    #/--New code--\
    if not source or not target:
        print("Node linking faliure: Source and/or target node not found:")
        print(source, "-->", target)
        print()
        return
    #\------------/

    self.node_tree.links.new(
        source.outputs[output],
        target.inputs[input]
    )
```

**Comment by NoneGiven** [comment]
Fixed in 0.26.1.0. Because the issue is with the `.dae` files, not the script, you'll need to re-export models with the new version (or change the slashes yourself in any that are already exported).
